src/carcockpit/shaders/shader_adv.hpp

```diff
@@ -22,25 +22,29 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
 #pragma once
 
 // #include <ruis/render/texturing_shader.hpp>
+#include <ruis/config.hpp>
 #include <ruis/render/opengles/shader_base.hpp>
 #include <ruis/render/texture_2d.hpp>
+#include <ruis/render/texture_cube.hpp>
 
-namespace ruis {
-using mat3 = r4::matrix3<real>;
-using matrix3 = mat3;
-static_assert(sizeof(mat3) == sizeof(real) * 3 * 3, "size mismatch");
-} // namespace ruis
-
-namespace carcockpit {
+namespace ruis::render {
 
 class shader_adv : public ruis::render::opengles::shader_base
 {
 public:
-	GLint sampler_normal_map, sampler_roughness_map;
-	GLint mat4_modelview, mat4_projection, mat3_normal;
-	GLint vec4_light_position, vec3_light_intensity, vec3_set_normal_mapping;
+	GLint sampler_normal_map;
+	GLint sampler_roughness_map;
+	GLint sampler_cube;
+
+	GLint mat4_modelview;
+	// GLint mat4_projection;
+	GLint mat3_normal;
 
-	ruis::vec3 set_normal_mapping_vector{1, 1, 1};
+	GLint vec4_light_position;
+	GLint vec3_light_intensity;
+	GLint vec3_set_normal_mapping;
+
+	ruis::vec3 set_normal_mapping_vector{1, 1, 1}; // experimental
```

Comment:
why is it experimental? Is it still in use? Remove if not.

Reply:
removed